libdnf5-plugins/CMakeLists.txt

```diff
@@ -4,3 +4,7 @@ set(CMAKE_C_VISIBILITY_PRESET hidden)
 add_subdirectory("actions")
 add_subdirectory("python_plugins_loader")
 add_subdirectory("rhsm")
+
+if(WITH_APPSTREAM)
+    add_subdirectory("appstream")
```

Comment:
It works, the Appstream data is installed as expected, with "one little caveat": when a session ends (on the dnf5daemon-server side), you also destroy the plugin class (which is fine), but you also unload the module (.so file). It breaks glib internals, because when another session opens and loads the plugin and calls the appstream library function, the glib aborts the dnf5daemon-server with:
```
(process:49835): GLib-GObject-CRITICAL **: 15:47:52.469: cannot register existing type 'AsMetadata'

(process:49835): GLib-GObject-CRITICAL **: 15:47:52.469: cannot add private field to invalid (non-instantiatable) type '<invalid>'

(process:49835): GLib-CRITICAL **: 15:47:52.469: g_once_init_leave_pointer: assertion 'result != 0' failed

(process:49835): GLib-GObject-CRITICAL **: 15:47:52.469: g_object_new_with_properties: assertion 'G_TYPE_IS_OBJECT (object_type)' failed
Segmentation fault (core dumped)
```

This does not exhibit when there's at least one session always loaded, which keeps the plugin module loaded in the memory, thus the glib type system is left in tact.

Is the plugin module unload really needed? I guess it can break more than just glib, it can break anything what stores some global data.

Reply:
I see. So this issue is related to the code that loads and unloads shared libraries [here](https://github.com/rpm-software-management/dnf5/blob/5.2.8.1/libdnf5/utils/library.cpp#L28-L38).

Could we simply solve this by passing `RTLD_NODELETE` during the library's `dlopen` to prevent unloading the libraries for the duration of the daemon's lifetime? Alternatively, should this be conditional for specific plugins only? Another approach could be to implement custom logic to unload libraries when the daemon terminates. @m-blaha @jrohel, could you please share your thoughts on this? @m-blaha @jrohel, could you please share your thoughts on this?